consensus/core/src/commit_consumer.rs

```
@@ -38,24 +39,66 @@ impl CommitConsumer {
 }
 
 pub struct CommitConsumerMonitor {
-    highest_handled_commit: AtomicU32,
+    // highest commit that has been handled by Sui
+    highest_handled_commit: watch::Sender<u32>,
+
+    // the highest commit found in local storage at startup
+    highest_observed_commit_at_startup: RwLock<u32>,
 }
 
 impl CommitConsumerMonitor {
     pub(crate) fn new(last_handled_commit: CommitIndex) -> Self {
         Self {
-            highest_handled_commit: AtomicU32::new(last_handled_commit),
+            highest_handled_commit: watch::Sender::new(last_handled_commit),
+            highest_observed_commit_at_startup: RwLock::new(0),
         }
     }
 
     pub(crate) fn highest_handled_commit(&self) -> CommitIndex {
-        self.highest_handled_commit
-            .load(std::sync::atomic::Ordering::Acquire)
+        *self.highest_handled_commit.borrow()
     }
 
     pub fn set_highest_handled_commit(&self, highest_handled_commit: CommitIndex) {
         self.highest_handled_commit
-            .store(highest_handled_commit, std::sync::atomic::Ordering::Release);
+            .send(highest_handled_commit)
+            .unwrap();
```

May need to ignore error on shutdown.